crypto/dilithium/sig_dilithium3.c

```diff
@@ -1,40 +1,56 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
-// SPDX-License-Identifier: Apache-2.0
-
+// SPDX-License-Identifier: Apache-2.0 OR ISC
 #include "../evp_extra/internal.h"
 #include "../fipsmodule/evp/internal.h"
 #include "sig_dilithium.h"
-#include "pqcrystals_dilithium_ref_common/api.h"
+#include "pqcrystals_dilithium_ref_common/sign.h"
+#include "pqcrystals_dilithium_ref_common/params.h"
+
+#include "./pqcrystals_dilithium_ref_common/fips202.c"
+#include "./pqcrystals_dilithium_ref_common/ntt.c"
+#include "./pqcrystals_dilithium_ref_common/packing.c"
+#include "./pqcrystals_dilithium_ref_common/params.c"
+#include "./pqcrystals_dilithium_ref_common/poly.c"
+#include "./pqcrystals_dilithium_ref_common/polyvec.c"
+#include "./pqcrystals_dilithium_ref_common/reduce.c"
+#include "./pqcrystals_dilithium_ref_common/rounding.c"
+#include "./pqcrystals_dilithium_ref_common/sign.c"
+#include "./pqcrystals_dilithium_ref_common/symmetric-shake.c"
 
 // Note: These methods currently default to using the reference code for
 // Dilithium. In a future where AWS-LC has optimized options available,
 // those can be conditionally (or based on compile-time flags) called here,
 // depending on platform support.
 
-
-int DILITHIUM3_keypair(uint8_t *public_key /* OUT */,
+int ml_dsa_65_keypair(uint8_t *public_key /* OUT */,
                        uint8_t *secret_key /* OUT */) {
-  return pqcrystals_dilithium3_ref_keypair(public_key, secret_key);
+  ml_dsa_params params;
+  ml_dsa_65_params_init(&params);
```

Comment:
This seems a bit inefficient, why not just define the 3 parameters statically and then pass in a reference to them instead of re-initializing a fresh params every time?

Reply:
Andrew and I met in person to discuss the details of this PR, I will note the discussions here:

The reason for creating these structures are to form some code consistency with how we deal with the same issue in ML-KEM. See https://github.com/aws/aws-lc/blob/main/crypto/fipsmodule/ml_kem/ml_kem_ref/params.c#L32-L40 for the respective part of the ML-KEM code. While this may not be the most efficient way to parse/point, I'd rather keep the code readable and somewhat analogous between the two algorithms. If we decide upon a design change for ML-DSA parameter selection, I'd like to see the same change in ML-KEM.